golem/managers/base.py

```diff
@@ -153,12 +153,14 @@ async def get_agreement(self) -> Agreement:
 
 class ActivityManager(Manager, ABC):
     @abstractmethod
-    async def do_work(self, work: Work) -> WorkResult:
+    async def get_activity(self) -> Activity:
         ...
 
 
 class WorkManager(Manager, ABC):
-    pass
+    @abstractmethod
+    async def do_work(self, work: Work) -> WorkResult:
+        ...
```

Comment:
Moving `do_work` to public interface for all work managers is not a good idea, because nearly every work manager will have different interface.